File: android/app/src/main/kotlin/com/typography/animations/entrance/Glitch.kt
package com.typography.animations.entrance

import com.typography.AnimatedTextState
import kotlin.math.cos

fun evaluateGlitch(t: Float): AnimatedTextState {
    if (t >= 1.0f) {
        return AnimatedTextState(
            opacity = 1f,
            offsetX = 0f,
            offsetY = 0f,
            scaleX = 1f,
            scaleY = 1f,
            rotation = 0f
        )
    }

    val p = 1.0f - t
    val offsetX = -0.25f * p * p
    val rotation = -15.0f * cos(t * 3.14159265f * 1.5f) * p

    return AnimatedTextState(
        opacity = t.coerceIn(0f, 1f),
        offsetX = offsetX,
        rotation = rotation
    )
}

File: lib/utils/animations/entrance/glitch.dart
import 'dart:math';
import '../../base/animated_text_state.dart';

AnimatedTextState evaluateGlitch(double t) {
  if (t >= 1.0) {
    return const AnimatedTextState(
      opacity: 1.0,
      offsetX: 0.0,
      offsetY: 0.0,
      scaleX: 1.0,
      scaleY: 1.0,
      rotation: 0.0,
    );
  }

  final p = 1.0 - t;
  final offsetX = -0.25 * p * p;
  final rotation = -15.0 * cos(t * pi * 1.5) * p; // Rocking pendulum swing

  return AnimatedTextState(
    opacity: t.clamp(0.0, 1.0),
    offsetX: offsetX,
    rotation: rotation,
  );
}
